### DISCUSSION TITLE
Best modus:ethernet adaptor?

### DISCUSSION THREAD

**Discussion by garethmarlow** [nathanmarlor/foxess_modbus#297] [General]
I can see that there are four modbus:ethernet/wifi adaptors which have been made to work, although there seems to be variation between them in effective polling frequency. I'm ready to buy one now - what would people recommend? I have an ethernet:wifi bridge already so an ethernet-only solution is fine, although one which did wifi natively would be neater.

**Comment by canton7** [comment]
Based on not very much, the Elfin ones (wired and wireless) look decent, have a good user interface, etc. The Waveshare are also decent, but they're a bit more industrial (and therefore possibly more complex?) and you'll need to source your own power supply, unless you go for PoE. I'd probably avoid the 410s based purely on how bad the interface is on the W610, although technically it appears to be fine.

**Comment by calum-mcfarlane** [comment]
FWIW, I have a 410S. The UI is not great, but then it doesn't need to be - you'll set it up once and then likely never touch it again. It's been very reliable, but then I'm sure the same can be said of many of this type of device. By modern standards, there just isn't that much to them.

That said, the Waveshare is (or was) significantly cheaper so you may as well get that one.

**Comment by garethmarlow** [comment]
Thanks for the swift replies. I've gone ahead with an Elfin (actually a Hyoduo) - Amazon can get me one by tomorrow, and if it doesn't work it can go back. Looking forward to trying out this integration - looks fantastic.

**Comment by garethmarlow** [comment]
Elfin/Hyoduo working well so far.

**Reply by dreampi** [reply]
How fast does it refresh? I have one coming this week

**Reply by garethmarlow** [reply]
I've just left things as the default setting. How would I tell how quickly it's refreshing?

**Reply by canton7** [reply]
We poll every 10s by default (configurable). You can enable debug logging to see how long polls take. There are also discussions on poll times, and I think there might be one on the Elfin specifically? On mobile, can check later.
